embassy-rp/src/spi.rs

```diff
@@ -346,7 +485,56 @@ mod eh1 {
     }
 }
 
-impl<'d, T: Instance> SetConfig for Spi<'d, T> {
+cfg_if::cfg_if! {
```

Comment:
use `#[cfg(...)] mod eha { ... }`. 

(Old code uses `cfg_if!` because rustc still parses code inside disabled cfg's, and Rust stable at that time couldn't parse the new GAT where-clause location. This is not the case anymore)